File: backend/core/domain/src/main/java/pl/lodz/p/it/core/domain/Account.java
package pl.lodz.p.it.core.domain;

import java.time.OffsetDateTime;
import java.util.HashSet;
import javax.validation.constraints.Email;
import javax.validation.constraints.Pattern;
import javax.validation.constraints.Positive;
import javax.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import org.hibernate.validator.constraints.Range;
import pl.lodz.p.it.core.shared.validation.RegexPattern;

/**
 * Class responsible for keeping a domain model of the account object.
 */
@EqualsAndHashCode(callSuper = true)
@Data
@Builder
@AllArgsConstructor
public class Account extends BaseModel {

    private String login;

    @Email(message = "Email is not valid")
    private String email;

    @Size(min = 60, max = 60)
    private String password;

    private Boolean active;

    private Boolean confirmed;

    private String firstName;

    private String lastName;

    @Pattern(regexp = RegexPattern.PHONE_NUMBER_PATTERN, message = "Invalid phone number")
    private String phoneNumber;

    @Size(max = 5)
    private String language;

    private String modifiedBy;

    private OffsetDateTime lastKnownGoodLogin;

    @Size(max = 39)
    private String lastKnownGoodLoginIp;

    private OffsetDateTime lastKnownBadLogin;

    @Size(max = 39)
    private String lastKnownBadLoginIp;

    @Range(min = 0)
    private Integer badLoginsCounter;

    @Positive
    private Float loyaltyFactor;

    private Boolean gymMember;

    private HashSet<TrainingPlan> trainingPlans;

    private HashSet<Diet> diets;
}

File: frontend/src/model/Account.js
export function Account(login, email, active, confirmed, firstName,
    lastName, phoneNumber, language, loyaltyFactor, gymMember,
    badLoginsCounter, modifiedBy, creationDate,
    modificationDate) {

  this.login = login;
  this.email = email;
  this.active = active;
  this.confirmed = confirmed;
  this.firstName = firstName;
  this.lastName = lastName;
  this.phoneNumber = phoneNumber;
  this.language = language;
  this.loyaltyFactor = loyaltyFactor;
  this.gymMember = gymMember;
  // this.lastKnownGoodLogin = lastKnownGoodLogin;
  // this.lastKnownGoodLoginIp = lastKnownGoodLoginIp;
  // this.lastKnownBadLogin = lastKnownBadLogin;
  // this.lastKnownBadLoginIp = lastKnownBadLoginIp;
  this.badLoginsCounter = badLoginsCounter;
  this.modifiedBy = modifiedBy;
  this.creationDate = creationDate;
  this.modificationDate = modificationDate;
}